### PR TITLE
Update go1.17

### PR DESCRIPTION
**Pull request by rminnich**

### CODE REVIEW

**Review by rjoleary — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # u-root
+TEST
```
Comment (rjoleary) [inline README.md:2]: Remove this line

**Review by rjoleary — APPROVED** [review]

**Comment by rminnich** [comment]
rminnich@a300:~/go/src/github.com/u-root/u-root$ go vet -composites=false ./cmds/... ./pkg/... ./
# github.com/u-root/u-root/pkg/spidev
pkg/spidev/spidev_linux_test.go:70:9: possible misuse of reflect.SliceHeader

what to do?

This warning got added ("should be harmless to warn people") at some point. I guess we don't count, since we actually use go vet for something. Today is the day for Go toolchain to push all my hot buttons.

**Comment by rminnich** [comment]
What's this shit about conflicts? There are non.

**Comment by rminnich** [comment]
https://github.com/golang/go/issues/40701

**Review by rjoleary — APPROVED** [review]

**Pull request merged**